server/packetio.go

```diff
@@ -48,19 +48,33 @@ const (
 	defaultWriterSize = 16 * 1024
 )
 
+// connBufferReader is a net.Conn compatible structure that reads from bufio.Reader
+// instead of the original net.Conn. It is used to support TLS.
+type connBufferReader struct {
```

Comment:
It's not a `Reader` interface any more, it also has `net.Conn` method.
So I think `bufferedConn` is a better name.